test/hooks.js

```diff
@@ -32,3 +34,49 @@ function createHookTest (hookName) {
 createHookTest('afterCopy')
 createHookTest('afterPrune')
 createHookTest('afterExtract')
+
+test('promisifyHooks executes functions in parallel', t => {
+  let output = '0'
+  const testHooks = [
+    done => { output += ' 1'; done() },
+    done => { setTimeout(() => { output += ' 2'; done() }, 1000) },
+    done => { output += ' 3'; done() },
+    done => { setTimeout(() => { output += ' 4'; done() }, 1000) },
+    done => { output += ' 5'; done() },
+    done => { setTimeout(() => { output += ' 6'; done() }, 1000) },
+    done => { output += ' 7'; done() },
+    done => { setTimeout(() => { output += ' 8'; done() }, 1000) },
+    done => { output += ' 9'; done() },
+    done => { setTimeout(() => { output += ' 10'; done() }, 1000) }
+  ]
+
+  return hooks.promisifyHooks(testHooks)
+    .then(() => t.not(output, '0 1 2 3 4 5 6 7 8 9 10', 'should not be in sequential order'))
+})
+
+test('serialHooks executes functions serially', t => {
+  let output = '0'
+  const timeoutFunc = number => {
```

Comment:
If this helper takes in the timeout as well we can remove more code duplication below for the hooks that instantly resolve

Reply:
:+1: https://github.com/electron-userland/electron-packager/pull/814/commits/9dcc68c1d0232382c3bd00d65602eca93d4213f5